pkg/quickwit/quickwit.go

```diff
@@ -26,6 +26,88 @@ type QuickwitDatasource struct {
 	dsInfo es.DatasourceInfo
 }
 
+type QuickwitMapping struct {
+	IndexConfig struct {
+		DocMapping struct {
+			TimestampField string `json:"timestamp_field"`
+			FieldMappings  []struct {
+				Name         string   `json:"name"`
+				InputFormats []string `json:"input_formats"`
```

Comment:
we want the `output_format`
the struct is a bit more complex, you can check the typescript code, particularly the `FieldMapping` type and the  `getAllFields` function.
Currently I still get a 500 with the log

```
grafana-quickwit-datasource  | logger=context userId=0 orgId=1 uname= t=2023-12-16T21:44:38.439604142Z level=error msg="Internal server error" error="[plugin.downstreamError] client: failed to call resources: failed to receive call resource response: rpc error: code = Unknown desc = {\"message\":\"Unmarshalling body error: err = json: cannot unmarshal array into Go value of type quickwit.QuickwitMapping, body = [\\n  {\\...
```

Reply:
Fixed: I use the `output_format` instead and flag it as optional.

So even if it doesn't exists, you shouldn't get unmarshalling errors anymore.

Tested with otel mapping:

<img width="1398" alt="Screenshot 2023-12-18 at 17 11 41" src="https://github.com/quickwit-oss/quickwit-datasource/assets/12884347/2df7421f-49c8-4194-96b4-59b16b097886">

The logs:

```shell
grafana-quickwit-datasource  | logger=plugin.quickwit-quickwit-datasource t=2023-12-18T16:11:34.091637818Z level=info msg="Found timestampFieldName = timestamp_nanos, timestampFieldFormat = unix_timestamp_nanos"
```

Reply:
Add recursivity for fields with an `object` type. No regression with the otel and yoi indices mappings:

<img width="910" alt="Screenshot 2023-12-22 at 15 07 50" src="https://github.com/quickwit-oss/quickwit-datasource/assets/12884347/59640310-26a8-414a-bab8-6d1da3708ad6">


```shell
grafana-quickwit-datasource  | logger=plugin.quickwit-quickwit-datasource t=2023-12-22T14:06:07.857021125Z level=info msg="Calling quickwit endpoint: http://host.docker.internal:7280/api/v1/indexes/otel-logs-v0_6"
grafana-quickwit-datasource  | logger=plugin.quickwit-quickwit-datasource t=2023-12-22T14:06:08.274010024Z level=info msg="Found timestampFieldName = timestamp_nanos, timestampFieldFormat = unix_timestamp_nanos"
```

Reply:
Test with this mapping:

```json
{
  "doc_mapping": {
    "dynamic_mapping": {
      "description": "the description",
      "expand_dots": true,
      "fast": true,
      "indexed": true,
      "record": "basic",
      "stored": true,
      "tokenizer": "raw"
    },
    "field_mappings": [
      {
        "name": "yo",
        "type": "object",
        "field_mappings": [{
          "name": "timestamp",
          "type": "datetime",
          "fast": true
        }]
      }
    ],
    "max_num_partitions": 1,
    "mode": "dynamic",
    "partition_key": "yo",
    "store_source": true,
    "tag_fields": [],
    "timestamp_field": "yo.timestamp"
  },
  "index_id": "yoii",
  "search_settings": {
    "default_search_fields": [
      "yo"
    ]
  },
  "version": "0.6"
}
```

And those documents:

```json
{"yo":{"timestamp": 1703255181}}
{"yo":{"timestamp": 1703255200}}
```

KO, I'll have to concatenate with the parent field name:

<img width="678" alt="Screenshot 2023-12-22 at 15 27 32" src="https://github.com/quickwit-oss/quickwit-datasource/assets/12884347/a04b7b56-2ffe-40f5-b0ac-2c077b822499">

Reply:
Fixed:

<img width="1094" alt="Screenshot 2023-12-22 at 15 33 47" src="https://github.com/quickwit-oss/quickwit-datasource/assets/12884347/0d82429e-56f0-4c5e-8ae7-31bc38d43527">

```shell
grafana-quickwit-datasource  | logger=plugin.quickwit-quickwit-datasource t=2023-12-22T14:33:37.941152856Z level=info msg="Calling quickwit endpoint: http://host.docker.internal:7280/api/v1/indexes/yoii"
grafana-quickwit-datasource  | logger=plugin.quickwit-quickwit-datasource t=2023-12-22T14:33:38.371661484Z level=info msg="Found timestampFieldName = yo.timestamp, timestampFieldFormat = rfc3339"
```